applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -654,6 +660,23 @@ static bool app_event_handler(const struct app_event_header *aeh)
 		return false;
 	}
 
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_SCAN_PM_EVENTS) &&
+	    is_power_down_event(aeh)) {
+		if (scanning) {
+			scan_stop();
```

Comment:
I think, as we check a condition is scanning is blocked right in scan_start() function, so we shouldn't care about works and other conditions, which could lead to scan start. It will not be started anyway. So work can execute as usual.

Reply:
I think that canceling the works and properly retriggering scanning on `wake_up_event` would be cleaner approach here. Then scanning would start instantly on `wake_up_event` (behavior is the same every time the module wakes up). With current approach the scanning start may be delayed by a random delay (depending on module's state during `wake_up_event` handling).
What do you think about it @pdunaj ?

Reply:
I think it's fine as it is. We actually should wait for ble to be functional so from logical point of view restarting scan on READY is fine I think.